### PR TITLE
Implementation of SSD - Single Shot Detector Head Component

### LINKED ISSUE #960: SSD - Single Shot Detector head component.
**Short Description**
Given that Object Detection is currently on the roadmap I think it would be great to also include an SSD Head component, which we could attach to different backbones for Object Detection task.

**Papers**
[SSD: Single Shot MultiBox Detector](https://arxiv.org/abs/1512.02325)

**Existing Implementations**
[Pytorch implementation](https://github.com/pytorch/vision/blob/8e0e715700877bd252836772704f513f7ed48265/torchvision/models/detection/ssd.py#L56)
Nvidia's [Pytorch Implementation](https://github.com/NVIDIA/DeepLearningExamples/blob/40eeae756c2c8b7f1d9140a5c846063802f5e6c8/PyTorch/Detection/SSD/ssd/model.py#L55)

### PR DESCRIPTION
**Pull request by srikesh-07**
# What this PR do ?

Fixes #960 

This PR implements the head component of SSD - Single Object Detector. 

The [Colab Notebook](https://colab.research.google.com/drive/1yi-seKkyRjbD9OwcYBdl8AvriPGB4U99?usp=sharing) could helps to verify the current implementation with Torchvision implementation.

Kindly check the PR and let me know if any changes want to be made.

Thank You.

### Reference Implementation:
[Torch Implementation](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/ssd.py#L58)

## Who can review?

@LukeWood @ianstenbit @tanzhenyu @sebastian-sz

### CODE REVIEW

File: keras_cv/layers/object_detection/ssd_head.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# Copyright 2023 The KerasCV Authors
```
Comment (innat) [inline keras_cv/layers/object_detection/ssd_head.py:1]: you may need to create separate directory for ssd. see others https://github.com/keras-team/keras-cv/tree/master/keras_cv/models/object_detection

Comment (srikesh-07) [reply keras_cv/layers/object_detection/ssd_head.py]: Fixed. Can you please review PR ?

**Review by srikesh-07 — COMMENTED** [review]
Files Reorganized as per above suggestion.

**Review by srikesh-07 — COMMENTED** [review]
Added `__init__.py` and reorganized files.

**Comment by jbischof** [comment]
@srikesh-07 sorry again for the slow reply but we're no longer moving forward on SSD

**Pull request closed without merging**